realm/src/main/java/io/realm/Realm.java

```diff
@@ -1115,11 +1110,22 @@ boolean contains(Class<?> clazz) {
      */
     public <E extends RealmObject> RealmResults<E> allObjectsSorted(Class<E> clazz, String fieldNames[],
                                                                boolean sortAscending[]) {
-        // FIXME: This is not an optimal implementation. When core's Table::get_sorted_view() supports
-        // FIXME: multi-column sorting, we can rewrite this method to a far better implementation.
-        RealmResults<E> results = this.allObjects(clazz);
-        results.sort(fieldNames, sortAscending);
-        return results;
+        io.realm.internal.android.Util.validateMultiSortParameters(fieldNames, sortAscending);
+
+        Table table = this.getTable(clazz);
+        List<TableView.Order> TVOrder = new ArrayList<TableView.Order>();
+        List<Long> columnIndices = new ArrayList<Long>();
```

Comment:
No need to create an List here, just use primitive array to avoid autoboxing later.